utility/getPipelineStages.py

```diff
@@ -0,0 +1,118 @@
+import json
+import logging
+import os
+import random
+import string
+import sys
+
+import yaml
+from docopt import docopt
+
+log = logging.getLogger(__name__)
+doc = """
+This script fetches all the tests to be run for a pipeline based on the RHCS version and overrides
+
+    Usage:
+        utility/getPipelineStages.py --rhcephVersion <VER> --tags <tags>
+                  [--overrides <str>]
+
+        utility/getPipelineStages.py (-h | --help)
+
+    Options:
+        -h --help          Shows the command usage
+        -v --rhcephVersion VER     The rhcephVersion for which test stages need to be fetched
+        -t --tags <str>    tags to be used for filtering test scripts
+        -o --overrides <str>       Overrides to be considered for execution
+"""
+
+
+def generate_random_string(length):
+    """Generate a random alphanumeric string of given length"""
+    return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+
+def fetch_stages(args):
+    """
+    Fetch test stages to be executed based on the input provided
+    Args:
+        args:
+            Dictionary containing the tier_level, cloud_type and all other necessary override parameters
+            to be used to fetch test stages for execution
+
+            arguments = {
+                "rhcephVersion": <the rhbuild version for which the stages need to be fetched>,
+                "tags": comma separated string containing the tags based on which the stages should be filtered,
+                "overrides": <overrides in json format, all keys accepted by run.py are supported to be overridden
+                            if any key similar to --store which does not have a value it can be passed as "store": ""
+            }
+
+            Example:
+                args = {
+                    "rhcephVersion": 5.1,
+                    "tags": "tier-1,stage-1,ibmc",
+                    "overrides": {"cloud_type": "psi", "platform": "rhel-8"}
+
+    Returns:
+        Yaml string containing CLI to be executed for each test suite at every stage
+        Examples:
+            "test-cephfs-core-features":
+                "execute_cli": ".venv/bin/python run.py ......",
+                 "cleanup_cli": "...."
+            .....
+    """
+    metadata_file = f"pipeline/metadata/{args['rhcephVersion']}.yaml"
```

Comment:
This could force us to execute the script from the root directory of the repo. If we could retrieve path of this file and then navigate to the metadata_file would allow us to execute from anywhere. That is

```
current_dir = os.path.dirname(os.path.abspath(__file__))
metadata_dir = os.path.abspath(f"{current_dir}/../pipeline/metadta")
```

Reply:
Done.